### PR TITLE
Show pending contacts

### PR DESCRIPTION
**Pull request by IanRDavies**

### CODE REVIEW

File: apps/android/app/src/main/java/chat/simplex/app/model/SimpleXAPI.kt [hunk 1]
```diff
@@ -339,6 +338,10 @@ open class ChatController(private val ctrl: ChatCtrl, private val ntfManager: Nt
         chatModel.updateNetworkStatus(r.contact, Chat.NetworkStatus.Connected())
 //        NtfManager.shared.notifyContactConnected(contact)
       }
+      is CR.ContactConnecting -> {
+        chatModel.updateContact(r.contact)
+        chatModel.updateNetworkStatus(r.contact, Chat.NetworkStatus.Connecting())
```
Comment (epoberezkin) [inline apps/android/app/src/main/java/chat/simplex/app/model/SimpleXAPI.kt:343]:
```suggestion
```

File: apps/android/app/src/main/java/chat/simplex/app/views/chatlist/ChatListNavLinkView.kt [hunk 2]
```diff
@@ -25,7 +25,12 @@ fun ChatListNavLinkView(chat: Chat, chatModel: ChatModel) {
       if (chat.chatInfo is ChatInfo.ContactRequest) {
         contactRequestAlertDialog(chat.chatInfo, chatModel)
       } else {
-        withApi { openChat(chatModel, chat.chatInfo) }
+        if (chat.chatInfo.ready) {
+          withApi { openChat(chatModel, chat.chatInfo) }
+        }
+        else {
```
Comment (epoberezkin) [inline apps/android/app/src/main/java/chat/simplex/app/views/chatlist/ChatListNavLinkView.kt:31]:
```suggestion
        } else {
```

Comment (epoberezkin) [reply apps/android/app/src/main/java/chat/simplex/app/views/chatlist/ChatListNavLinkView.kt:31]: let's please not use this style...

File: apps/android/app/src/main/java/chat/simplex/app/model/SimpleXAPI.kt (hunk 1 above)
Comment (IanRDavies) [reply apps/android/app/src/main/java/chat/simplex/app/model/SimpleXAPI.kt:343]: If we drop this we can also crop the `Connecting` object entirely

File: apps/ios/Shared/Model/ChatModel.swift [hunk 3]
```diff
@@ -382,12 +382,14 @@ final class Chat: ObservableObject, Identifiable {
     enum NetworkStatus: Decodable, Equatable {
         case unknown
         case connected
+        case connecting
```
Comment (epoberezkin) [inline apps/ios/Shared/Model/ChatModel.swift:385]:
```suggestion
```

File: apps/ios/Shared/Views/ChatList/ChatListNavLink.swift [hunk 4]
```diff
@@ -53,11 +54,18 @@ struct ChatListNavLink: View {
         }
         .swipeActions(edge: .trailing) {
             Button(role: .destructive) {
-                AlertManager.shared.showAlert(deleteContactAlert(contact))
+                if contact.ready {
+                    AlertManager.shared.showAlert(deleteContactAlert(contact))
+                }
+                else {
+                    AlertManager.shared.showAlert(deletePendingContactAlert(chat, contact))
+                }
```
Comment (epoberezkin) [inline apps/ios/Shared/Views/ChatList/ChatListNavLink.swift:62]:
```suggestion
                AlertManager.shared.showAlert(
                    contact.ready
                    ? deleteContactAlert(contact)
                    : deletePendingContactAlert(chat, contact)
                )
```

File: apps/ios/Shared/Views/ChatList/ChatListNavLink.swift [hunk 5]
```diff
@@ -53,11 +54,18 @@ struct ChatListNavLink: View {
...
             } label: {
                 Label("Delete", systemImage: "trash")
             }
         }
+        .onTapGesture { showPendingContactAlert = true && !contact.ready }
+        .alert(isPresented: $showPendingContactAlert) { pendingContactAlert(chat, contact) }
```
Comment (epoberezkin) [inline apps/ios/Shared/Views/ChatList/ChatListNavLink.swift:68]: can also be done via alert manager.

Comment (epoberezkin) [inline apps/ios/Shared/Views/ChatList/ChatListNavLink.swift:68]:
```suggestion
        .onTapGesture {
            if !contact.ready {
                AlertManager.shared.showAlert(pendingContactAlert(chat, contact))
            }
        }
```

File: apps/ios/Shared/Views/ChatList/ChatListNavLink.swift [hunk 6]
```diff
@@ -12,6 +12,7 @@ struct ChatListNavLink: View {
     @EnvironmentObject var chatModel: ChatModel
     @State var chat: Chat
     @State private var showContactRequestDialog = false
+    @State private var showPendingContactAlert = false
```
Comment (epoberezkin) [inline apps/ios/Shared/Views/ChatList/ChatListNavLink.swift:15]:
```suggestion
```

**Review by epoberezkin — APPROVED** [review]

**Pull request merged**